src/WorldMap.cpp

```diff
@@ -1,33 +1,228 @@
+	: m_position(position),
+	m_tileID(tileID),
+	m_tileSheet(tileSheet)
+{
+	m_sprite.setTexture(m_tileSheet.getTexture());
+	m_sprite.setTextureRect(m_tileSheet.getTileLocationByID(m_tileID));
+	const int tileSize = m_tileSheet.getDetails().m_tileSize;
+	m_sprite.setPosition(sf::Vector2f(m_position.x * tileSize, m_position.y * tileSize));
+}
+
+WorldMap::TileLayer::Tile::Tile(const Tile& orig)
+	: m_position(orig.m_position),
+	m_tileID(orig.m_tileID),
+	m_tileSheet(orig.m_tileSheet)
+{
+	m_sprite.setTexture(m_tileSheet.getTexture());
+	m_sprite.setTextureRect(m_tileSheet.getTileLocationByID(m_tileID));
+	const int tileSize = m_tileSheet.getDetails().m_tileSize;
+	m_sprite.setPosition(sf::Vector2f(m_position.x * tileSize, m_position.y * tileSize));
+}
+
+WorldMap::TileLayer::TileLayer(const std::vector<std::vector<int>>& tileMapData, const sf::Vector2i& mapSize, const TileSheet& tileSheet, std::string&& name)
+	: m_name(std::move(name)),
+	m_tileMap()
+{
+	loadInTileMap(tileMapData, mapSize, tileSheet);
+}
+
+void WorldMap::TileLayer::loadInTileMap(const std::vector<std::vector<int>>& tileMapData, const sf::Vector2i& mapSize, const TileSheet& tileSheet)
+{
+	if (!m_tileMap.empty())
+	{
+		clearMap();
+	}
+
+	m_tileMap.reserve(static_cast<size_t>(mapSize.x * mapSize.y));
+
+	for (int col = 0; col < mapSize.y; ++col)
+	{
+		for (int row = 0; row < mapSize.x; ++row)
+		{
+			int tileDrawID = tileMapData[col][row]; //Get ID for tile
+			if (tileDrawID > 0)
+			{
+				m_tileMap.emplace_back(sf::Vector2i(row, col), tileSheet, tileDrawID);
+			}
+		}
+	}
 }
 
-void WorldMap::assignTileLayer(const std::vector<std::vector<int>>& tileData, const sf::Vector2i & mapSize, std::string&& tileLayerName)
+void WorldMap::TileLayer::draw(sf::RenderWindow& window)
 {
-	if (!hasTileLayer(tileLayerName))
+	for (const auto& tile : m_tileMap)
 	{
-		m_tileLayers.emplace_back(tileData, mapSize, m_tileSheetDetails, std::move(tileLayerName), m_mapTexture);
+		window.draw(tile.m_sprite);
 	}
 }
 
+void WorldMap::TileLayer::clearMap()
+{
+	m_tileMap.clear();
+}
+
+const std::string& WorldMap::TileLayer::getName() const
+{
+	return m_name;
+}
+
+//WorldMap.cpp
+WorldMap::WorldMap(const std::string& mapName)
+	: m_tileLayers()
+{
+	parseLevel(mapName);
+}
+
+WorldMap::~WorldMap()
+{
+}
+
+void WorldMap::assignTileLayer(const std::vector<std::vector<int>>& tileData, const sf::Vector2i & mapSize, const std::string& tileSheetName, std::string&& tileLayerName)
+{
+	assert(!hasTileLayer(tileLayerName));
+	m_tileLayers.emplace_back(tileData, mapSize, getTileSheet(tileSheetName), std::move(tileLayerName));
+}
+
 bool WorldMap::hasTileLayer(const std::string& tileLayerName) const
 {
 	return (std::find_if(m_tileLayers.cbegin(), m_tileLayers.cend(),
 		[&tileLayerName](auto& tileLayer) { return tileLayer.getName() == tileLayerName; }) != m_tileLayers.cend() ? true : false);
 }
 
+bool WorldMap::hasTileSheet(const std::string & tileSheetName) const
```

Comment:
This function is unused, so I doubt it should be kept.  If for some reason it sticks around, let's change the body to be a bit more concise: